src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonConfigService.kt

```diff
@@ -39,4 +52,28 @@ class PrisonConfigService(
   fun getSupportedPrisons(): List<String> {
     return prisonRepository.getSupportedPrisons()
   }
+
+  @Transactional
+  fun createPrison(prisonDto: PrisonDto): PrisonDto {
+    val newPrison = Prison(prisonDto.code, prisonDto.active)
+    val savedPrison = prisonRepository.saveAndFlush(newPrison)
+    val excludeDates = prisonDto.excludeDates.map { PrisonExcludeDate(prisonId = savedPrison.id, prison = savedPrison, it) }
+    savedPrison.excludeDates.addAll(excludeDates)
+
+    return mapEntityToDto(savedPrison)
+  }
+
+  @Transactional
+  fun activatePrison(prisonCode: String): PrisonDto {
+    val prisonToUpdate = findPrisonByCode(prisonCode)
+    prisonToUpdate.active = true
+    return mapEntityToDto(prisonToUpdate)
```

Comment:
do we not need a repository call here.

Reply:
nope save when exits the transaction scope. (JPA proxy)

Reply:
actually your right